packages/contracts/src/contracts/current/protocol/Exchange/MixinSettlementProxy.sol

```diff
@@ -20,22 +20,24 @@ pragma solidity ^0.4.21;
 pragma experimental ABIEncoderV2;
 
 import "./mixins/MSettlement.sol";
-import "../TokenTransferProxy/ITokenTransferProxy.sol";
 import "../../tokens/Token/IToken.sol";
 import "./LibPartialAmount.sol";
+import "../AssetProxyDispatcher/IAssetProxy.sol";
 
 /// @dev Provides MixinSettlement
 contract MixinSettlementProxy is
     MSettlement,
     LibPartialAmount
 {
-
-    ITokenTransferProxy TRANSFER_PROXY;
+    IAssetProxy TRANSFER_PROXY;
+    // TODO: Is there a way to encode ZRX_PROXY_DATA into takerAssetData/makerAssetData?
+    //       As it stands, only one of the ERC20 proxies can be used.
+    bytes ZRX_PROXY_DATA;
```

Comment:
At launch will it be the new ERC20 proxy, or the old? If it is the old proxy would this force all of the new order makers to set the allowance? 

We can't infer the assetProxyId for ZRX based off the assetProxyId in the respective maker and taker metadata since it is now more than ERC20. I guess we're in a safe position setting this to the new ERC20 proxy since there are currently no fees being forced by relayers.

Reply:
We chatted about this offline, and decided the old ERC20 proxy will be removed for simplicity. The new ERC20 proxy will be used here.